SU-HSDO/suhumsci
# STN-362: Card Image Utilities

# READY FOR REVIEW

## Summary
This PR adds a set of responsive utility classes to customize the width of images in card components. Classes can be applied to horizontal cards, vertical cards, and structured cards. Classes must be applied to the entire view (not just the image) and will follow a similar naming convention to the [Decanter Grid System](https://decanter.stanford.edu/page/layouts-grid-system/):
- `hb-card-image-4-of-12` sets the image to 1/4th the width of the card at all screen sizes
- `hb-card-image-md-4-of-12` sets the image to 1/4th the width of the card at the medium break point and larger

These classes have been documented in the H&S Pattern Documentation.

There should be no change to any existing card patterns, the change is purely additive.

## Need Review By (Date)
05/18/2020

## Urgency
medium

## Steps to Test
1. Confirm that all tests pass
2. Confirm that all card patterns have not changed in their default state
3. Modify a view and apply the following classes to the entire view:
    `hb-card-image-12-of-12 hb-card-image-sm-6-of-12 hb-card-image-md-3-of-12`
    <img width="1751" alt="Custom_Econ_News__Content____Department_of_Economics" src="https://user-images.githubusercontent.com/2486846/82070804-4209de00-96a3-11ea-9507-05e2a33a529d.png">
    This should cause the width of the image to be 100% of the card at the smallest screen sizes then change to 50% at the small breakpoint, and then be 25% of the width at the medium breakpoint.


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
